### PR TITLE
feat: Display warning on agent connection timeout

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Screen Shot 2022-11-09 at 14 36 20

### CODE REVIEW

File: site/src/components/Resources/AgentStatus.tsx [hunk 1]
```diff
@@ -50,7 +56,52 @@ const ConnectingStatus: React.FC = () => {
...
+  agent: WorkspaceAgent
+  workspace: Workspace
+}> = ({ agent, workspace }) => {
+  const { t } = useTranslation("agent")
+  const styles = useStyles()
+  const anchorRef = useRef<SVGSVGElement>(null)
+  const [isOpen, setIsOpen] = useState(false)
+  const id = isOpen ? "timeout-popover" : undefined
+  const troubleshootLink =
+    agent.troubleshooting_url ?? `/templates/${workspace.template_name}#readme`
```
Comment (mafredri) [inline site/src/components/Resources/AgentStatus.tsx:69]: The `#readme` didn't cause my browser (Safari) to scroll the readme into view, should that happen?

File: site/src/i18n/en/agent.json [hunk 2]
```diff
@@ -9,5 +9,14 @@
     "os": "OS",
     "apps": "Apps",
     "noApps": "None"
-  }
+  },
+  "status": {
+    "timeout": "Timeout"
+  },
+  "timeoutTooltip": {
+    "title": "Agent taking too long to connect",
```
Comment (mafredri) [inline site/src/i18n/en/agent.json:17]:
```suggestion
    "title": "Agent is taking too long to connect",
```

**Review by mafredri — APPROVED** [review]
Looks great functionality wise 👍🏻. The exclamation icon looks a bit weird in my eyes but I don't have a suggestion to make it better. 😄

I did notice that I had to manually refresh the workspace page for the agent status to update. But it looks like my session didn't have a SSE session open so maybe that's why. Not sure why it isn't working.

File: site/src/components/Resources/AgentStatus.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/components/Resources/AgentStatus.tsx:69]: Fixed!

**Comment by BrunoQuaresma** [comment]
@mafredri 

> Looks great functionality wise 👍🏻. The exclamation icon looks a bit weird in my eyes but I don't have a suggestion to make it better. 😄

Hm... maybe it is the color or the icon itself? 

> I did notice that I had to manually refresh the workspace page for the agent status to update. But it looks like my session didn't have a SSE session open so maybe that's why. Not sure why it isn't working.

Probably. Since the status is updated automatically.

**Review by presleyp — APPROVED** [review]

**Pull request merged**